internal/manifests/collector/ingress.go

```diff
@@ -75,22 +66,76 @@ func Ingress(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemet
 			},
 		},
 		Spec: networkingv1.IngressSpec{
-			TLS: otelcol.Spec.Ingress.TLS,
-			Rules: []networkingv1.IngressRule{
-				{
-					Host: otelcol.Spec.Ingress.Hostname,
-					IngressRuleValue: networkingv1.IngressRuleValue{
-						HTTP: &networkingv1.HTTPIngressRuleValue{
-							Paths: paths,
-						},
+			TLS:              otelcol.Spec.Ingress.TLS,
+			Rules:            rules,
+			IngressClassName: otelcol.Spec.Ingress.IngressClassName,
+		},
+	}
+}
+
+func createPathIngressRules(otelcol string, hostname string, ports []corev1.ServicePort) networkingv1.IngressRule {
+	pathType := networkingv1.PathTypePrefix
+	paths := make([]networkingv1.HTTPIngressPath, len(ports))
+	for i, port := range ports {
+		portName := naming.PortName(port.Name, port.Port)
+		paths[i] = networkingv1.HTTPIngressPath{
+			Path:     "/" + port.Name,
+			PathType: &pathType,
+			Backend: networkingv1.IngressBackend{
+				Service: &networkingv1.IngressServiceBackend{
+					Name: naming.Service(otelcol),
+					Port: networkingv1.ServiceBackendPort{
+						Name: portName,
 					},
 				},
 			},
-			IngressClassName: otelcol.Spec.Ingress.IngressClassName,
+		}
+	}
+	return networkingv1.IngressRule{
+		Host: hostname,
+		IngressRuleValue: networkingv1.IngressRuleValue{
+			HTTP: &networkingv1.HTTPIngressRuleValue{
+				Paths: paths,
+			},
 		},
 	}
 }
 
+func createSubdomainIngressRules(otelcol string, hostname string, ports []corev1.ServicePort) []networkingv1.IngressRule {
+	var rules []networkingv1.IngressRule
+	pathType := networkingv1.PathTypeExact
+	for _, port := range ports {
+		portName := naming.PortName(port.Name, port.Port)
+
+		host := fmt.Sprintf("%s.%s", portName, hostname)
+		if hostname == "" || hostname == "*" {
+			host = portName
```

Comment:
wouldnt this make the host something like "otlp" instead of "otlp.example.com"? should we just error out here?

Reply:
it makes sense to error. I will add a validation to the webhook for it